app/src/main/java/de/kabiskac/weekdaybug/ui/main/Fragment1.kt

```diff
@@ -24,9 +25,11 @@ class Fragment1 : Fragment() {
             inflater,
             R.layout.fragment1, container, false
         )
-        viewModel = ViewModelProvider(this).get(MainViewModel::class.java)
+        viewModel = ViewModelProvider(requireActivity()).get(MainViewModel::class.java)
```

Comment:
You need to share the same lifecycle owner to get the same ViewModel instance. This uses the activity hosting the navigation graph as the shared lifecycle owner.